xml/System.Diagnostics.Tracing/EventSource.xml

```diff
@@ -82,6 +92,28 @@
  :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR/etwtrace/cs/program.cs" id="Snippet1":::
  :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR/etwtrace/vb/program.vb" id="Snippet1":::  
   
+
+## Advanced Usage
+ Traditionally, user defined <xref:System.Diagnostics.Tracing.EventSource>s expect to inherit directly from <xref:System.Diagnostics.Tracing.EventSource>. For advanced scenarios, however, users can create `abstract` <xref:System.Diagnostics.Tracing.EventSource>s, called a Utility Source, and implement interfaces. Using one or both of these techniques allows users to share code between different derived sources.
+
+> [!IMPORTANT]
+>  Abstract <xref:System.Diagnostics.Tracing.EventSource>s cannot define keywords, tasks, opcodes, channels, or events.
+
+> [!IMPORTANT]
+>  When using interfaces with <xref:System.Diagnostics.Tracing.EventSource>, do not explicitly implement interface methods to avoid name collisions at runtime when generating event metadata.
```

Comment:
```suggestion
> To avoid name collisions at run time when generating event metadata, don't explicitly implement interface methods when using interfaces with <xref:System.Diagnostics.Tracing.EventSource>.
```